### PR TITLE
Fix the kebab menu close when we click outside the kebab.

### PR DESCRIPTION
**Pull request by TimothyAsirJeyasing**
BZ: https://bugzilla.redhat.com/show_bug.cgi?id=2308314
https://issues.redhat.com/browse/DFBUGS-351

### CODE REVIEW

File: packages/shared/src/kebab/kebab.tsx [hunk 1]
```diff
@@ -22,6 +23,24 @@ import { ModalKeys, defaultModalMap } from '../modals/types';
...
+  callback: () => void
+) => {
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (ref.current && !ref.current.contains(event.target as Node)) {
+        callback();
+      }
+    };
+
+    document.addEventListener('mousedown', handleClickOutside);
```
Comment (alfonsomthd) [inline packages/shared/src/kebab/kebab.tsx:37]: Provided we want to mimic the behavior of the OCP console, you also have to close the kebab menu when the user presses the `Escape` key as [the console does](https://github.com/openshift/console/blob/master/frontend/public/components/utils/kebab.tsx#L98) (e.g. go to `Compute > Nodes` to check the behavior of the kebab when the ESC key is pressed).

cc @SanjalKatiyar @bipuladh @GowthamShanmugam

**Comment by alfonsomthd** [comment]
LGTM. @SanjalKatiyar for tagging.

**Comment by alfonsomthd** [comment]
LGTM. @SanjalKatiyar for tagging.

File: packages/shared/src/kebab/kebab.tsx (hunk 1 above)
Comment (bipuladh) [reply packages/shared/src/kebab/kebab.tsx:43]: This makes sense. Let's update this accordingly.

**Review by bipuladh — APPROVED** [review]
/lgtm

**Comment by SanjalKatiyar** [comment]
/lgtm cancel
small comment

File: packages/shared/src/kebab/kebab.tsx [hunk 2]
```diff
@@ -102,18 +129,18 @@ export const Kebab: React.FC<KebabProps> & KebabStaticProperties = ({
...
-
   const eventRef = React.useRef(undefined);
   const [toggleDirection, setToggleDirection] =
     React.useState<DropdownPopperProps['direction']>('down');
   const [isOpen, setOpen] = React.useState(false);
+  const dropdownRef = React.useRef<HTMLDivElement>(null); // Ref for the dropdown menu
 
-  const { resourceModel, resource } = extraProps;
+  // Use the custom hook to detect clicks outside the Kebab menu
+  useClickOutside(dropdownRef, () => setOpen(false));
```
Comment (SanjalKatiyar) [inline packages/shared/src/kebab/kebab.tsx:140]: u are passing callback `() => setOpen(false)` to dep list of useEffect inside `useClickOutside` hook, but ref of this callback is changing on each re-render.

Comment (SanjalKatiyar) [reply packages/shared/src/kebab/kebab.tsx:140]: wrap it around `useCallback` plz...

**Comment by SanjalKatiyar** [comment]
@TimothyAsirJeyasing plz add corresponding Jira BZ link to this PR...

**Comment by alfonsomthd** [comment]
@TimothyAsirJeyasing please rebase & solve conflicts.

**Pull request merged**